src/views/view_sycl.hpp

```diff
@@ -232,32 +249,60 @@ struct MatrixView<
   SYCL_BLAS_INLINE scalar_t *get_pointer() const { return ptr_; }
 
   /**** EVALUATING ***/
-  SYCL_BLAS_INLINE scalar_t &eval(index_t indx) { return *(ptr_ + indx); }
 
-  SYCL_BLAS_INLINE const scalar_t eval(index_t indx) const noexcept {
-    return *(ptr_ + indx);
+  SYCL_BLAS_INLINE scalar_t &eval(index_t i, index_t j) {
+    return ((layout::is_col_major()) ? *(ptr_ + i + sizeL_ * j)
+                                     : *(ptr_ + j + sizeL_ * i));
   }
 
-  SYCL_BLAS_INLINE scalar_t &eval(index_t i, index_t j) {
+  SYCL_BLAS_INLINE scalar_t eval(index_t i, index_t j) const noexcept {
     return ((layout::is_col_major()) ? *(ptr_ + i + sizeL_ * j)
                                      : *(ptr_ + j + sizeL_ * i));
   }
 
+  template <bool use_as_ptr = false>
+  SYCL_BLAS_INLINE typename std::enable_if<!use_as_ptr, scalar_t &>::type eval(
+      index_t indx) {
+    const index_t j = indx / sizeR_;
+    const index_t i = indx - sizeR_ * j;
+    return eval(i, j);
+  }
+
+  template <bool use_as_ptr = false>
+  SYCL_BLAS_INLINE typename std::enable_if<!use_as_ptr, scalar_t>::type
+  eval(index_t indx) const noexcept {
+    const index_t j = indx / sizeR_;
+    const index_t i = indx - sizeR_ * j;
+    return eval(i, j);
+  }
+
   SYCL_BLAS_INLINE scalar_t &eval(cl::sycl::nd_item<1> ndItem) {
     return eval(ndItem.get_global_id(0));
   }
 
-  SYCL_BLAS_INLINE const scalar_t eval(cl::sycl::nd_item<1> ndItem) const
+  SYCL_BLAS_INLINE scalar_t eval(cl::sycl::nd_item<1> ndItem) const
       noexcept {
     return eval(ndItem.get_global_id(0));
   }
 
+  template <bool use_as_ptr = false>
```

Comment:
Should these not be moved above so they are below their non-enable_if versions?

Reply:
The non `enable_if` versions are a different overload. The one with `enable_if` has `enable_if` both for the `true` and `false` cases